Src/FluentAssertions/Equivalency/Selection/ExcludeMemberByPathSelectionRule.cs

```diff
@@ -17,17 +16,20 @@ public ExcludeMemberByPathSelectionRule(MemberPath pathToExclude)
             memberToExclude = pathToExclude;
         }
 
-        protected override IEnumerable<SelectedMemberInfo> OnSelectMembers(IEnumerable<SelectedMemberInfo> selectedMembers,
-            string currentPath, IMemberInfo context)
+        protected override void AddOrRemoveMembersFrom(List<IMember> selectedMembers, INode parent, string parentPath, MemberSelectionContext context)
         {
-            return selectedMembers
-                .Where(memberInfo => !memberToExclude.IsSameAs(new MemberPath(memberInfo.DeclaringType, currentPath.Combine(memberInfo.Name))))
-                .ToArray();
+            foreach (IMember member in selectedMembers.ToArray())
```

Comment:
If you like, it could be written as
```
selectedMembers.RemoveAll(member => memberToExclude.IsSameAs(new MemberPath(member.DeclaringType, parentPath.Combine(member.Name))));
```